### ISSUE TITLE
Extend request methods for Download button

### ISSUE THREAD

**Issue by drwelby** [radiantearth/stac-browser#356]
Just had a chance to test the Auth functionality and it works wonderfully.

However, the Download button for an asset doesn't pass on the bearer token so the request is rejected.

Is it possible to include the Auth functionality on asset links, at least when the asset and API are on the same host?

**Comment by drwelby** [comment]
This also seems to apply to the "Show on Map" button if you're doing client-side COG rendering.

**Comment by drwelby** [comment]
This is the same domain, does that option apply then?

**Comment by m-mohr** [comment]
Yes, because without whitelisting it only sends private information to things that are underneath the configured catalogUrl.

**Comment by drwelby** [comment]
URLs are

Catalog URL:

`https://api.maxar.com/discovery/v1`

Asset URL:
`https://api.maxar.com/discovery/v1/collections/wv03-vnir/items/1040010087416200/assets/collections/dg-archive/assets/browse/1040010087416200.browse.tif`

So it should count as a child item.

**Comment by m-mohr** [comment]
Yes. What form of providing the token do you use? Query or Header?
Because this only works for query parameters in download as otherwise it's just a normal link that you click in a Browser, which doesn't support appending headers.
The show on map I need to check separately. Could be that stac-layer doesn't take over the headers, too. For query parameters it should work.

**Comment by drwelby** [comment]
It's type `header`.

**Comment by m-mohr** [comment]
Yeah, that's tricky then. The browser can't easily send headers to the download request and I'm pretty sure stac-layer also doesn't handle headers. So that needs improvements in stac-layer (and OL STAC for the migration at some point). If there's any chance to switch to query parameters I'd recommend that for now, otherwise this is something that we'd need to work on but have no funding for right now.

**Comment by drwelby** [comment]
I figured it was tricky but wasn't sure if there was some Vue component that could do it.

Query based authentication exists but isn't integrated into this API at this time.

**Comment by m-mohr** [comment]
Some basic investigations:
- For Download try the HTML5 File API ([example 1](https://stackoverflow.com/questions/24501358/how-to-set-a-header-for-a-http-get-request-and-trigger-file-download/24523253#24523253), [example 2](https://gist.github.com/nerdyman/5de9cbe640eb1fbe052df43bcec91fad))
- For Show on Map:
  - stac-layer
    - Client-side Probably needs some work in the underlying libraries such as georaster-layer-for-leaflet. [Issue](https://github.com/stac-utils/stac-layer/issues/66)
    - Tile-Server: not possible AFAIK
  - OL STAC:
    - Client-side COG: Support passing HTTP headers with the requests (via the GeoTiffSourceOptions)
    - Tile-Server: not possible AFAIK

**Comment by m-mohr** [comment]
There's a potential fix for the Download button in PR #429.
Could you test it please @drwelby ?
